plugins/completion/cmp/sources/_mk-cmp-plugin.nix

```diff
@@ -0,0 +1,32 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+{
+  name,
+  sourceName,
+  defaultPackage ? pkgs.vimPlugins.${name},
+  maintainers ? [ lib.maintainers.GaetanLepage ],
+  imports ? [ ],
+  # Could be set to mkNeovimPlugin
+  pluginFn ? helpers.vim-plugin.mkVimPlugin,
+  ...
+}@args:
+pluginFn config (
+  builtins.removeAttrs args [
+    "package"
+    "pluginFn"
+    "sourceName"
+  ]
```

Comment:
TBH, I think being able to do more "complex" plugin definitions in sources.nix is probably overkill.

I think the most we'd need there is `package` and `maintainers`?

Anything more complex could be moved _out of_ sources.nix and declare its cmp-source association itself.

Though this would mean we'd need to rethink how the "all sources" _test_ works...

Reply:
Although not currently in use, I've kept passing in the module args, as even basic stuff like setting a different maintainer or package would need that.

Reply:
Actually, just moving the list into `sources/default.nix` makes more sense IMO. This is what we to for language-servers, and since we don't need to read the sources list in the test anymore it simplifies things IMO.